plugins/modules/oneview_label.py

```diff
@@ -0,0 +1,146 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: oneview_label
+short_description: Manage OneView Label resources.
+description:
+    - Provides an interface to manage Label resources. Can create, update, or delete.
+version_added: "2.5.0"
+requirements:
+    - "python >= 2.7.9"
+    - "hpeOneView >= 6.1.0"
+    - "ansible >= 2.9"
+author: "Asis Bagga (@asisbagga)"
+options:
+    state:
+        description:
+            - Indicates the desired state for the label resource.
+              C(present) will ensure data properties are compliant with OneView.
+              C(absent) will remove the resource from OneView, if it exists.
+        choices: ['present', 'absent']
+        type: str
+        required: true
+    data:
+        description:
+            - List with Label properties.
+        required: true
+        type: dict
+
+extends_documentation_fragment:
+    - hpe.oneview.oneview
+    - hpe.oneview.oneview.validateetag
+    - hpe.oneview.oneview.params
+'''
+
+EXAMPLES = '''
+- name: Create Labels for enclosure reosurces
+  oneview_label:
+    config: "{{ config }}"
+    state: present
+    data:
+      resourceUri: "/rest/enclosures/0000000000A66102"
+      labels:
+        - name: "Test label 1"
+        - name: "Test Label 2"
+  delegate_to: localhost
+  register: label
+- debug: var=label
+
+- name: Update label of given reosurce for enclosure reosurces
+  oneview_label:
+    config: "{{ config }}"
+    state: present
+    data:
+      resourceUri: "/rest/enclosures/0000000000A66102"
+      labels:
+        - name: "Test label 1 Renamed"
+          uri: null
+        - name: "Test label 2 Renamed"
+          uri: null
+        - name: "Test label 3"
+          uri: null
+  delegate_to: localhost
+  register: label
+- debug: var=label
+
+- name: Delete Labels for enclosure reosurces
+  oneview_label:
+    config: "{{ config }}"
+    state: absent
+    data:
+      resourceUri: "/rest/enclosures/0000000000A66102"
+  delegate_to: localhost
+  register: label
+- debug: var=label
+'''
+
+RETURN = '''
+label:
+    description: Has the facts about the OneView label.
+    returned: On state 'present'. Can be null.
+    type: dict
+'''
+
+from ansible_collections.hpe.oneview.plugins.module_utils.oneview import OneViewModule
+
+
+class LabelModule(OneViewModule):
+    MSG_CREATED = 'Label created successfully.'
+    MSG_UPDATED = 'Label updated successfully.'
+    MSG_DELETED = 'Label deleted successfully.'
+    MSG_ALREADY_PRESENT = 'Label is already present.'
+    MSG_ALREADY_ABSENT = 'Label is already absent.'
+    RESOURCE_FACT_NAME = 'Labels'
```

Comment:
Labels fact name should be lowercase ```labels``` to maintain consistency

Reply:
done